datastore/google/cloud/datastore/key.py

```diff
@@ -427,3 +485,153 @@ def _validate_project(project, parent):
             raise ValueError("A Key must have a project set.")
 
     return project
+
+
+def _urlsafe_b64decode(urlsafe):
+    """Replacement for base64.urlsafe_b64decode.
+
+    Borrowed from ``ndb.key._DecodeUrlSafe``, noting their comment:
+
+        This is 3-4x faster than urlsafe_b64decode()
+
+    :type urlsafe: bytes
+    :param urlsafe: The encoded ASCII string to be decoded
+
+    :rtype: bytes
+    :returns: The value that was decoded.
+    """
+    # This is 3-4x faster than urlsafe_b64decode()
+    return base64.b64decode(
```

Comment:
ew micro-optimizations. Does this actually differ in behavior at all? If not, can we just use the normal `urlsafe_b64decode` and `encode`?